menas/src/main/scala/za/co/absa/enceladus/menas/repositories/MonitoringMongoRepository.scala

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.repositories
+
+import org.mongodb.scala._
+//import org.mongodb.scala.{AggregateObservable, Completed, Document, MapReduceObservable, MongoDatabase}
+//import org.mongodb.scala.bson.BsonDocument
+//import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Aggregates._
+import org.mongodb.scala.model.Filters._
+//import org.mongodb.scala.model.{FindOneAndUpdateOptions, ReturnDocument, Updates}
+import org.mongodb.scala.Document
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Repository
+import za.co.absa.enceladus.model.Run
+
+import scala.concurrent.Future
+
+object MonitoringMongoRepository {
+  val collectionName = "run"
+}
+
+@Repository
+class MonitoringMongoRepository @Autowired()(mongoDb: MongoDatabase)
+  extends MongoRepository[Run](mongoDb) {
+  private[menas] override def collectionName: String = MonitoringMongoRepository.collectionName
+
+
+  def getMonitoringDataPoints(datasetName: String, startDate: String, endDate: String): Future[Seq[String]] = {
+    // scala mongodb driver does not yet support all mql features, so we use Document() with aggregate pipelines
+    val observable: AggregateObservable[Document] = collection
+      .aggregate(Seq(
+        // filter by dataset name
+        filter(equal("dataset", datasetName)),
+        //add casted dates
+        Document("""{$addFields: {
```

Comment:
Any chance we could use the scala DSL for these?

Reply:
[062bda1](https://github.com/AbsaOSS/enceladus/pull/453/commits/062bda173c82c040e0faa3f46cda690d487ec31c) Switched to native mongodb scala driver methods where possible. It does not support needed Date and array expressions yet, so those parts had to remain.